script.js

```diff
@@ -1,25 +1,76 @@
-var appKey = "721c180b09eb463d9f3191c41762bb68",
+var appKey = '721c180b09eb463d9f3191c41762bb68',
     logsStarted = false,
     engagementData = {},
+    authenticationData = {},
     getEngagementMaxRetries = 10,
     chatWindow,
     chatContainer,
     chat,
     chatState,
     chatArea,
-    logsLastChild;
+    logsLastChild,
+    ssoKey,
+    AUTH0_CLIENT_ID = 'zX55x13W00Tqa6mUSxuzRojUGpUcd27E',
+    AUTH0_DOMAIN = 'livepersonauth.auth0.com';
 
 initDemo();
 
+window.authenticateMethod = function(func) {
+    func({ ssoKey, redirect_uri: location.href });
+};
+
 function initDemo() {
-    if (lptag === "true") {
+    if (lptag === 'true') {
         createExternalJsMethodName();
     }
     else {
         initChat(getEngagement);
     }
+    initAuth0();
+
+    if(!ssoKey) {
+        var code = getURLParams(window.location.search).code;
+        if(code) {
+            ssoKey = code;
+        }
+    }
 }
 
+function initAuth0() {
+    $('docuemnt').ready(function() {
+        var lock = new Auth0Lock(AUTH0_CLIENT_ID, AUTH0_DOMAIN, {
+            auth: {
+                redirectUrl: location.href,
+                responseType: 'code',
+                params: {
+                    scope: 'openid'
+                }
+            }
+        });
+
+        lock.on('authenticated', function(authResult) {
+            writeLog(`authentication - ${JSON.stringify(authResult)}`);
+            ssoKey = null;
+            console.log(authResult);
+            if (authResult && (authResult.accessToken || authResult.idToken)) {
+                ssoKey = authResult.idToken || authResult.accessToken;
```

Comment:
This is looking for a JWT or oAuth2 Bearer. But the above initAuth0 expects an authKey (OpenID code).